### ISSUE TITLE
Classic block still displaying unwanted p and br tags when Edit as HTML is chosen

### ISSUE THREAD

**Issue by bobbingwide** [WordPress/gutenberg#10837]
**Describe the bug**
Classic block still displaying unwanted p and br tags when Edit as HTML is chosen. These are somewhat misleading. These unwanted/added tags are not saved in the post_content during update.

**To Reproduce**
1. Using the Classic editor create a post something like this.
```
Classic block still displaying unwanted p and br tags when Edit as HTML is chosen. These are somewhat misleading. These unwanted/added tags are not saved in the post_content during update.
<!--more-->

[clear]

[bw_csv]A,B,C
1,2,3
4,5,6
[/bw_csv]

end of post
```
2. Edit using Gutenberg and see the contents in the Classic block
3. Choose More Options > Edit as HTML
4. Notice `p` and `br` tags being added.
5. Edit the text and perform an Update.
6. View the post.
7. The unwanted tags have magically disappeared. The content is actually how I want it.

**Expected behavior**
The Edit as HTML display should not be showing the p and br tags. It's misleading.

**Screenshots**

The first screen shot is step 4.
![image](https://user-images.githubusercontent.com/2474435/47258500-dabe8480-d493-11e8-8ef0-58b7a389055b.png)

The second screen shot is step 7. You can see that I'd edited the content.
In the sidebar you can see the raw post_content that has been saved as wanted. 

![image](https://user-images.githubusercontent.com/2474435/47258521-148f8b00-d494-11e8-8bbe-41c4b0c53975.png)

**Desktop (please complete the following information):**
 - OS: [e.g. iOS] Windows 10
 - Browser [e.g. chrome, safari] Chrome
 - Version [e.g. 22] Version 69.0.3497.100 (Official Build) (64-bit)

**Additional context**
- This is a follow up to #5123. 
- The Classic block in v4.0.0 is significantly better than when I first reported the problem in v2.6 or so

**Comment by tisayama** [comment]
I have the same problem. Does anyone have an update?

**Comment by danicholls** [comment]
Is this still a problem?

**Comment by bobbingwide** [comment]
Yes.  
![36B60B4C-3E0E-4A37-A8CD-AF7882DA1C68](https://user-images.githubusercontent.com/2474435/54981004-2fb07a80-4f9f-11e9-812f-b3b0c1c9e606.jpeg)

**Comment by paaljoachim** [comment]
Hey @bobbingwide Herb
Can you retest with the newest version of WordPress and Gutenberg?
Thanks!

**Comment by bobbingwide** [comment]
Yes. Same result if I follow my original instructions.
```
<p>Classic block still displaying unwanted p and br tags when Edit as HTML is chosen. These are somewhat misleading. These unwanted/added tags are not saved in the post_content during update.<br /><!--more--></p>
<p>[clear]</p>
<p>[bw_csv]A,B,C<br />1,2,3<br />4,5,6<br />[/bw_csv]</p>
<p>end of post</p>
<p>perform update.</p>
```

**Comment by t-hamano** [comment]
This issue is still reproducible:

https://github.com/user-attachments/assets/d3d528f8-0303-47be-8f6f-d8465ac49e5a

**Comment by annezazu** [comment]
Can still replicate in 6.8.2.

**Comment by paaljoachim** [comment]
George @Mamaduka Is this something you can perhaps take a look at?
Thank you.

**Comment by Mamaduka** [comment]
It seems that the additional HTML is inserted when the block is parsed, even though the block specifies the content source to be `raw`. I don't see the same behavior with the Custom HTML block, which uses a similar source type.

I'll see if I can track down the exact cause, but this isn't a high priority for the moment.

**Update:** I think I've found the source. It looks like `autop` is intentionally applied to the Freeform content. It's been like this since forever #4005. Changing this can have some unexpected side effects.

https://github.com/WordPress/gutenberg/blob/95108ddbf24f6f67533de27affeb4e91a306f6ff/packages/blocks/src/api/parser/index.js#L99-L108

## Screenshot

<img width="1576" height="191" alt="Image" src="https://github.com/user-attachments/assets/a5bfeda8-14e7-4299-8691-ff8bdc0bc236" />

**Comment by Mamaduka** [comment]
A one solution I could think of is to remove `autop` when editing Classic editor content as HTML and add it back when changes are done. This primarily works as expected, but formatting is lost after switching to visual and requires an editor refresh.

What's the current reason for using Classic block? It was always meant to be a temporary block when switching from classic to the block editor.

@aduth, @youknowriad, I know it's been years since you worked on the original PR (#4005), but I would appreciate your input.

## Screencast

https://github.com/user-attachments/assets/fc78b9e3-e0bc-4559-8db1-2aa84f1f2c8c

**Comment by aduth** [comment]
My memory is fuzzy on the details, but as best I can recall:

>What's the current reason for using Classic block? It was always meant to be a temporary block when switching from classic to the block editor.

For compatibility reasons, the editor needed to be able to generate the same original content from pre-Gutenberg posts to ensure that the value in the database didn't unexpectedly change, particularly if the user didn't actually meaningfully edit content, and if they relied on the newline-delimited content from before Gutenberg (like in widget examples earlier in the issue). I don't know that we could ever get rid of that requirement to preserve this type of content, since we can assume there will always be the possibility of someone editing a post with content written pre-Gutenberg.

It gets trickier because this behavior only applies if there's no blocks in the post. The block _does_ have `<p>` tags (and saves with them) if there are other blocks in the post, so it's sometimes correct for the block to show them when editing as HTML.

In how it's applied, the difference is:

- Post content saving goes through [`__unstableSerializeAndClean`](https://github.com/WordPress/gutenberg/blob/bd02787d2d61fb3a4830f2813b08331fc8523dec/packages/blocks/src/api/serializer.js#L397-L418), which is what handles the `removep` before saving
- The "edit as HTML" uses [`getBlockInnerHTML`](https://github.com/WordPress/gutenberg/blob/bd02787d2d61fb3a4830f2813b08331fc8523dec/packages/blocks/src/api/serializer.js#L300-L325), which doesn't apply this `removep` behavior

Maybe `getBlockInnerHTML` should be updated to incorporate this to be truthful to what's actually saved, but it's tricky with how it depends on there being other blocks in the content.

Another option could be to disable [`html` supports](https://developer.wordpress.org/block-editor/reference-guides/block-api/block-supports/#html) for freeform blocks? It's a bit of a hacky way to address the issue 😅

**Comment by Mamaduka** [comment]
> Another option could be to disable [html supports](https://developer.wordpress.org/block-editor/reference-guides/block-api/block-supports/#html) for freeform blocks? It's a bit of a hacky way to address the issue.

We had to disable a lot of standard features, which don't work for various reason with Classic. The list also includes `html` editing.

I understand this isn't the best solution, but due to nature of classic block the listed features don't function as expected. It was always recommended to migrage legacy content to blocks.

Closing this as "won't fix".

**Comment by tyxla** [comment]
I agree this should be closed, and closing now for real as @Mamaduka intended earlier.
